File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/utils/ZeppelinLogger.scala
package org.intellij.plugin.zeppelin.utils

import org.intellij.plugin.zeppelin.idea.toolwindow.TextLogger


object ZeppelinLogger {
  var output: ZeppelinOutput = new TextLogger

  def initOutput(output: ZeppelinOutput): Unit = this.output = output

  def printError(msg: String): Unit = output.printError("[ERROR] " + msg)

  def printMessage(msg: String): Unit = output.printMessage(if (msg.endsWith("\n")) msg.dropRight(1) else msg)

  def printSuccess(msg: String): Unit = output.printMessage("[SUCCESS] " + msg)
}


trait ZeppelinOutput {
  def printError(msg: String)

  def printMessage(msg: String)
}

File: zeppelin/implementation/src/org/intellij/plugin/zeppelin/utils/ZeppelinLogger.kt
package org.intellij.plugin.zeppelin.utils

import org.intellij.plugin.zeppelin.idea.toolwindow.TextLogger

object ZeppelinLogger {
    var output: ZeppelinOutput = TextLogger()

    fun printError(msg: String): Unit = output.printError("[ERROR] $msg")
    fun printMessage(msg: String): Unit = output.printMessage(if (msg.endsWith("\n")) msg.dropLast(1) else msg)
    fun printSuccess(msg: String): Unit = output.printMessage("[SUCCESS] $msg")
}

interface ZeppelinOutput {
    fun printError(msg: String)
    fun printMessage(msg: String)
}